tests/scripts/all.sh

```diff
@@ -2758,15 +2758,21 @@ component_test_zeroize () {
 
 component_test_psa_compliance () {
     msg "build: make, default config (out-of-box), libmbedcrypto.a only"
-    make library/libmbedcrypto.a
+    make -C library libmbedcrypto.a
 
     msg "unit test: test_psa_compliance.py"
     ./tests/scripts/test_psa_compliance.py
 }
 
 support_test_psa_compliance () {
-    local ver=($(cmake --version | sed 's/cmake version //; y/./ /; q'))
-    [ "${ver[0]}" -eq 3 ] && [ "${ver[1]}" -ge 10 ]
+    ver="$(cmake --version)"
+    ver="${ver#cmake version }"
+    ver_major="${ver%%.*}"
+
+    ver="${ver#*.}"
+    ver_minor="${ver%%.*}"
+
+    [ "$ver_major" -eq 3 ] && [ "$ver_minor" -ge 10 ]
```

Comment:
Note: this will reject CMake 4.0.0.

Reply:
Yes. I was under the impression that CMake was using Semantic versioning as well - [but that doesn't strictly seem to be the case](https://cmake.org/cmake/help/latest/variable/CMAKE_VERSION.html).
I can change it to accept 4.0.0+ as well, if you'd like.

Reply:
Hmm, I think you're right, let's keep it that way, so when 4.0.0 comes out we get a clean error, while if we accept 4.0 and they make incompatible changes the errors might be much more cryptic.

Reply:
CMake 3.0 is (mostly?) compatible with 2.x, so I don't think we should reject CMake 4.

We could make CMake itself do the version comparison. Hopefully, if CMake 4 is incompatible with 3.x, they'll design their APIs so that old scripts will error out.
```
cmake -P /dev/stdin <<<"cmake_minimum_required(VERSION 3.101)"
```

Reply:
>`cmake -P /dev/stdin <<<"cmake_minimum_required(VERSION 3.10)"`

This works in `bash`, but not in `dash` (relevant for the 2.x backport), because `dash` implements heredocs as a pipe, as opposed to bash, which writes them to a temporary file. CMake requires scripts to be seekable, so only bash's approach works.

Reply:
`all.sh` uses bash in 2.28 as well. Only 2.16 is staying with sh.